vars/getBaseHardwareConfig.groovy

```diff
@@ -0,0 +1,16 @@
+def call(vm_networks, vm_hardware_memory_mb, vm_hardware_num_cpus) {
+
+  def nets = vm_networks.inject([]) { result, network ->
+                                      result << "  - {${ network.collect().join(', ') }}"
+                                    }.join('\n').replaceAll("=", ": ")
```

Comment:
Nice! Although it feels a little bit too magic. I'd propose:
1. Create a helper function (in this file) eg. `mapToYaml` that converts a `Map` to yaml syntax (basically `network.collect().join(', ')`
2. Change inject to just `collect` and simply `join('')` (so move `\n` from `join` to the closure)
3. Move `replaceAll` to `mapToYaml`, so it's more obvious what generates the `=` which we're converting

Reply:
We have here list of maps, each map represents network. As far as I know using ```join``` change whole collection into String so basically you cannot use ```join``` twice on nested collection.

Reply:
Now you have one `join`inside the closure and one outside. I didn't mean to change that

Reply:
But by using ```inject``` method I still have a list, just containing desired strings generated from maps. Without that I would have one string and second operation ```join``` would not succeed.